### DISCUSSION TITLE
Terminal command to open a layout

### DISCUSSION THREAD

**Discussion by horverno** [foxglove/community#26] [Q&A] [answered]
Hi,

I wonder if there is a possibility to open a config / layout and connect to a data source from a single command in terminal such as:
``` r
foxglove-studio --layout demo1.json --source http://localhost:11311
```
This would be extremely handy for several use cases. In e.g. rviz there is a similar function:
```r
rosrun rviz rviz -d demo1.rviz
```
I have seen `injectFilesToOpen: no files - skipping` in the terminal, but I did not found anything useful about that, I don't even know if this is related at all.

Many thanks!

**Comment by amacneil** [comment, marked as answer]
You can do this using the deep link functionality described here:

https://foxglove.dev/docs/studio/building-and-sharing-links

I believe you can pass any `foxglove://open` URL as a parameter to the CLI if you wish to open a specific data source or layout from the command line.

**Comment by horverno** [comment]
Thanks @amacneil 
I tried the following:
`$ firefox -new-tab "https://studio.foxglove.dev/?ds=sample-nuscenes&openIn=desktop"` >> it works, but I need to click to allow open Foxglove
`$ firefox -new-tab "foxglove://open?ds=sample-nuscenes"` >> if I click to the tab and press enter, this opens up

Is it possible to avoid the extra clicks?

**Reply by amacneil** [reply]
Try just running `foxglove-studio "foxglove://open?ds=sample-nuscenes"`. You should be able to pass a URL as the first argument to the binary, without involving Firefox.

Let me know if that works, we should add it to the deep link docs page.

**Comment by horverno** [comment]
Thanks @amacneil it works :+1:  
As a side note i noticed a typo in this description: https://foxglove.dev/docs/studio/building-and-sharing-links

``` bash
foxglove-studio "foxglove://open?ds=ros1-socket&ds.url=localhost:11311"
```
 should be 
``` bash
foxglove-studio "foxglove://open?ds=ros1-socket&ds.url=http://localhost:11311"
```
(with `http://`)

Also can I somehow use `layoutId` locally?

**Reply by amacneil** [reply]
Thank you, doc fix incoming!

You can only reference layouts that exist in your account or in your browser storage. It is not currently possible to reference a local json file.

**Reply by juliangaal** [reply]
As far as I can tell, this is still the case. Are there any plans you know of that address opening local json files?
